jgrapht-ext/src/main/java/org/jgrapht/ext/CSVExporter.java

```diff
@@ -0,0 +1,334 @@
+        }
+        this.delimiter = delimiter;
+        this.parameters = new HashSet<>();
+    }
+
+    /**
+     * Exports a graph
+     *
+     * @param g the graph
+     * @param writer the writer
+     */
+    public void exportGraph(Graph<V, E> g, Writer writer)
+    {
+        PrintWriter out = new PrintWriter(writer);
+        switch (format) {
+        case EDGE_LIST:
+            exportAsEdgeList(g, out);
+            break;
+        case ADJACENCY_LIST:
+            exportAsAdjacencyList(g, out);
+            break;
+        case MATRIX:
+            exportAsMatrix(g, out);
+            break;
+        }
+        out.flush();
+    }
+
+    /**
+     * Return if a particular parameter of the exporter is enabled
+     * 
+     * @param p the parameter
+     * @return {@code true} if the parameter is set, {@code false} otherwise
+     */
+    public boolean isParameter(CSVFormat.Parameter p)
+    {
+        return parameters.contains(p);
+    }
+
+    /**
+     * Set the value of a parameter of the exporter
+     * 
+     * @param p the parameter
+     * @param value the value to set
+     */
+    public void setParameter(CSVFormat.Parameter p, boolean value)
+    {
+        if (value) {
+            parameters.add(p);
+        } else {
+            parameters.remove(p);
+        }
+    }
+
+    /**
+     * Get the format of the exporter
+     * 
+     * @return the format of the exporter
+     */
+    public CSVFormat getFormat()
+    {
+        return format;
+    }
+
+    /**
+     * Set the format of the exporter
+     * 
+     * @param format the format to use
+     */
+    public void setFormat(CSVFormat format)
+    {
+        this.format = format;
+    }
+
+    /**
+     * Get the delimiter (comma, semicolon, pipe, etc).
+     * 
+     * @return the delimiter
+     */
+    public char getDelimiter()
+    {
+        return delimiter;
+    }
+
+    /**
+     * Set the delimiter (comma, semicolon, pipe, etc).
+     * 
+     * @param delimiter the delimiter to use
+     */
+    public void setDelimiter(char delimiter)
+    {
+        if (!DSVUtils.isValidDelimiter(delimiter)) {
+            throw new IllegalArgumentException(
+                "Character cannot be used as a delimiter");
+        }
+        this.delimiter = delimiter;
+    }
+
+    private void exportAsEdgeList(Graph<V, E> g, PrintWriter out)
+    {
+        for (E e : g.edgeSet()) {
+            String s = DSVUtils.escapeDSV(
+                vertexIDProvider.getVertexName(g.getEdgeSource(e)),
+                delimiter);
+            String t = DSVUtils.escapeDSV(
+                vertexIDProvider.getVertexName(g.getEdgeTarget(e)),
+                delimiter);
+            out.println(s + delimiter + t);
+        }
+    }
+
+    private void exportAsAdjacencyList(Graph<V, E> g, PrintWriter out)
+    {
+        if (g instanceof DirectedGraph<?, ?>) {
+            for (V v : g.vertexSet()) {
+                out.print(vertexIDProvider.getVertexName(v));
```

Comment:
Shouldn't this be:
out.print(
                     DSVUtils.escapeDSV(
                         vertexIDProvider.getVertexName(v),
                         delimiter));